docs/notebooks/deep_learning/resnet_flax.md

```diff
@@ -130,76 +181,73 @@ def augment(image, label):
-    image, label = next(raw_train_ds_iter)
-    augmented_image, _ = augment(image, label)
-    axes[i, j].imshow(tf.cast(augmented_image, tf.uint8), cmap=plt.cm.gray_r, interpolation="nearest")
-    axes[i, j].set_axis_off()
-    axes[i, j].set_title(info.features['label'].names[label], fontsize=10, y=0.9)
+
+train_loader_augmented = train_loader.map(augment)
+plot_sample_images(train_loader_augmented)
 ```
 
-Finally, on the test set, we won't be applying data augmentation to keep a reliable evaluation of the model's performance. In the test set we'll only normalize the pixels to be in the range [0, 1].
++++ {"id": "ITnYDuiIDBkM"}
 
-```{code-cell} ipython3
-def normalize(image, label):
-  image = tf.cast(image, tf.float32)
-  image = (image / 255.0)
-  return image, label
+Finally, we shuffle the data in the train set and create batches of size `'BATCH_SIZE'` for both train and test set
 
-train_ds = raw_train_ds.map(augment).map(normalize)
-test_ds = raw_test_ds.map(normalize)
+```{code-cell} ipython3
+:id: TSjMCbukDBkN
 
-train_ds = train_ds.shuffle(5000, reshuffle_each_iteration=True).batch(
-    BATCH_SIZE, drop_remainder=True).prefetch(tf.data.AUTOTUNE).repeat(2 * MAX_EPOCH).as_numpy_iterator()
-test_ds = test_ds.shuffle(5000).batch(
-    BATCH_SIZE, drop_remainder=True).prefetch(tf.data.AUTOTUNE).repeat(MAX_EPOCH).as_numpy_iterator()
+train_loader_batched = train_loader_augmented.shuffle(
+    buffer_size=10_000, reshuffle_each_iteration=True
+).batch(BATCH_SIZE, drop_remainder=True)
 
+test_loader_batched = test_loader.batch(BATCH_SIZE, drop_remainder=True)
+```
 
-NUM_CLASSES = info.features["label"].num_classes
++++ {"id": "CrSSsvpUDBkN"}
 
-IMG_SIZE = info.features["image"].shape
-```
+With the data ready, we can now define the model. Below we define the ResNet architecture that we'll later instantiate. We define different variants of the architecture with different sizes and depths (`'ResNet1'`, `'ResNet18'`, `'ResNet34'`, `'ResNet50'` and `'ResNet101'`).
 
-With the data ready, we can now define the model. Below we define the ResNet architecture that we'll later instantiate. We define different variants of the architecture with different sizes and depths (`'ResNet1'`, `'ResNet18'`, `'ResNet34'`, `'ResNet50'` and `'ResNet101'`). The architecture is based on the [Flax imagenet example](https://github.com/google/flax/blob/main/examples/imagenet/models.py).
+The following code is based on the [Flax imagenet example](https://github.com/google/flax/blob/main/examples/imagenet/models.py). The only difference with that code is the addition of the keyword argument `initial_conv_config` to the `ResNet` class, which allows to change the configuration of the initial convolutional layer of the network. This is important to get state of the art accuracy on CIFAR10, as the default kernel size (7, 7) is too big for the small 32x32 images of CIFAR10.
 
 ```{code-cell} ipython3
-:id: onkVLRw7L3j4
+:id: P7Z3Vex8QuGz
+
+ModuleDef = Any
```

Comment:
What do we gain from this definition? It wasn't entirely clear to me while reading the code. Maybe add a quick comment explaining why you introduce this.

Reply:
TBH I'm doing it because flax does it that way (https://github.com/google/flax/blob/main/examples/imagenet/models.py). I guess `conv: ModuleDef` is more explicit (for a human) than `conv: Any` , although the parser will treat them equally?

Reply:
If that's the Flax way, fine with me!